lib/mapObjects/CGTownInstance.cpp

```diff
@@ -487,72 +503,73 @@ void CGTownInstance::newTurn(CRandomGenerator & rand) const
-
+		
 		//get Mana Vortex or Stables bonuses
 		//same code is in the CGameHandler::buildStructure method
 		if (visitingHero != nullptr)
 			cb->visitCastleObjects(this, visitingHero);
-
+		
 		if (garrisonHero != nullptr)
 			cb->visitCastleObjects(this, garrisonHero);
-
+		
 		if (tempOwner == PlayerColor::NEUTRAL) //garrison growth for neutral towns
+		{
+			std::vector<SlotID> nativeCrits; //slots
+			for(const auto & elem : Slots())
 			{
-				std::vector<SlotID> nativeCrits; //slots
-				for(const auto & elem : Slots())
+				if (elem.second->type->getFaction() == subID) //native
 				{
-					if (elem.second->type->getFaction() == subID) //native
-					{
-						nativeCrits.push_back(elem.first); //collect matching slots
-					}
+					nativeCrits.push_back(elem.first); //collect matching slots
 				}
-				if(!nativeCrits.empty())
+			}
+			if(!nativeCrits.empty())
+			{
+				SlotID pos = *RandomGeneratorUtil::nextItem(nativeCrits, rand);
+				StackLocation sl(this, pos);
+				
+				const CCreature *c = getCreature(pos);
+				if (rand.nextInt(99) < 90 || c->upgrades.empty()) //increase number if no upgrade available
 				{
-					SlotID pos = *RandomGeneratorUtil::nextItem(nativeCrits, rand);
-					StackLocation sl(this, pos);
-
-					const CCreature *c = getCreature(pos);
-					if (rand.nextInt(99) < 90 || c->upgrades.empty()) //increase number if no upgrade available
-					{
-						cb->changeStackCount(sl, c->getGrowth());
-					}
-					else //upgrade
-					{
-						cb->changeStackType(sl, VLC->creh->objects[*c->upgrades.begin()]);
-					}
+					cb->changeStackCount(sl, c->getGrowth());
+				}
+				else //upgrade
+				{
+					cb->changeStackType(sl, VLC->creh->objects[*c->upgrades.begin()]);
 				}
-				if ((stacksCount() < GameConstants::ARMY_SIZE && rand.nextInt(99) < 25) || Slots().empty()) //add new stack
+			}
+			if ((stacksCount() < GameConstants::ARMY_SIZE && rand.nextInt(99) < 25) || Slots().empty()) //add new stack
+			{
+				int i = rand.nextInt(std::min(GameConstants::CREATURES_PER_TOWN, cb->getDate(Date::MONTH) << 1) - 1);
+				if (!town->creatures[i].empty())
 				{
-					int i = rand.nextInt(std::min(GameConstants::CREATURES_PER_TOWN, cb->getDate(Date::MONTH) << 1) - 1);
-					if (!town->creatures[i].empty())
-					{
-						CreatureID c = town->creatures[i][0];
-						SlotID n;
-
-						TQuantity count = creatureGrowth(i);
-						if (!count) // no dwelling
-							count = VLC->creh->objects[c]->getGrowth();
-
-						{//no lower tiers or above current month
-
-							if ((n = getSlotFor(c)).validSlot())
-							{
-								StackLocation sl(this, n);
-								if (slotEmpty(n))
-									cb->insertNewStack(sl, VLC->creh->objects[c], count);
-								else //add to existing
-									cb->changeStackCount(sl, count);
-							}
+					CreatureID c = town->creatures[i][0];
+					SlotID n;
+					
+					TQuantity count = creatureGrowth(i);
+					if (!count) // no dwelling
+						count = VLC->creh->objects[c]->getGrowth();
```

Comment:
Here we absolutely can use `VLC->creatures()->getByIndex(c)->getGrowth()`. Can you use it instead of direct calling of VLC->creh?

Reply:
I suppose it's safe to do, but refactoring of CGTownInstance is out of scope. Change here is just dixing indents, see above code is the same

Reply:
It is a new code, so, we do not need to use constructa than need refactoring in new code.

Reply:
Okay, but I suggest to change it regardless, because then we will have less work for refactoring.